mantidproject/mantid
# MergeRuns sum option and error control

Adds the following changes:
 1. Sum merge type - this allows a sample log to be simply summed
 2. Sample logs are now overwritten, and not renamed
 3. Option to skip a file if binning is different
 4. Option to throw an error instead of skipping

**To test:**

Load 2 data files that make sense to merge.

 1. Try setting a sum property in `MergeRuns`, either via the algorithm or instrument parameters file, check the output is as expected
 2. Sample logs should be overwitten with new type, should be noticable for TimeSeries and List options. Check a TimeSeries/List is created instead of any existing numeric type. Check merging two workspaces that were already merged once also works.
 3. Rebin one workspace, check that workspace is skipped in the merge
 4. Check the algorithm throws when using `RebinBehaviour=Skip` and `FailBehaviour=Stop`. Also check for the `SampleLogsFail` option and `FailBehaviour=Stop`.

Also try a few unscripted tests, and check code looks good.

Fixes #19042 
Fixes #17520 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
